python/mxnet/gluon/data/dataloader.py

```diff
@@ -363,13 +449,8 @@ def __init__(self, dataset, batch_size=None, shuffle=False, sampler=None,
 
     def __iter__(self):
```

Comment:
Would it make sense to add documentation about the API of returned `_SameProcessIter` / `_MultiWorkerIter`? Ie. that they will reset when `__iter__` is called on them.